zio-config/src/main/scala/zio/config/PropertyTree.scala

```diff
@@ -0,0 +1,11 @@
+package zio.config
+
+sealed trait PropertyTree[+K, +V]
```

Comment:
I'd also add `flatten`, we can do lots others too, `mapKey`, `map` / `mapValue`, `fold`, etc.